nano/node/active_transactions.cpp

```diff
@@ -68,98 +68,155 @@ void nano::active_transactions::stop ()
 void nano::active_transactions::block_cemented_callback (std::shared_ptr<nano::block> const & block_a)
 {
 	auto transaction = node.store.tx_begin_read ();
+	auto status_type = election_status (transaction, block_a);
 
-	boost::optional<nano::election_status_type> election_status_type;
-	if (!confirmation_height_processor.is_processing_added_block (block_a->hash ()))
+	if (!status_type)
+		return;
+
+	switch (*status_type)
+	{
+		case nano::election_status_type::inactive_confirmation_height:
+			process_inactive_confirmation (transaction, block_a);
+			break;
+
+		default:
```

Comment:
I don't think we should use a default clause here but that's the existing functionality so I guess we can keep it.